app/src/main/java/org/mozilla/fenix/search/SearchInteractor.kt

```diff
@@ -55,6 +59,13 @@ class SearchInteractor(
 
     override fun onTextChanged(text: String) {
         store.dispatch(SearchAction.UpdateQuery(text))
+
+        if (startedTyping.ranOnTextChanged && !startedTyping.userHasTyped) {
+            store.dispatch(SearchAction.ShowSearchShortcutEnginePicker(false))
+            turnOnStartedTyping()
```

Comment:
Why not just do `startedTyping.userHasTyped = true` here? The line below it already handles `ranOnTextChanged = true`

Reply:
Ah, I didn't see you use this function elsewhere. If your purpose is to be explicit that this is considered the user "turning on typing" I actually prefer this :) The named function makes it clear what's happening here. Ignore me!